src/components/split_button/split_button.tsx

```diff
@@ -0,0 +1,225 @@
+     * Applied to the outermost wrapper (popover)
+     */
+    popoverClassName?: string;
+
+    /**
+     * Controls whether the options are shown upon initial render. Default: false
+     */
+    initiallyOpen?: boolean;
+
+    /**
+     * Optional additional props to send to Primary Button
+     */
+    propsForPrimaryButton?: OuiButtonProps;
+
+    /**
+     * Optional additional props to send to Dropdown Button
+     */
+    propsForDropdownButton?: OuiButtonProps;
+
+    /**
+     * Optional additional props to send to each Option Item, when
+     * it is string, rendered with OuiText wrapper
+     */
+    propsForOptionItems?: OuiTextProps;
+  };
+
+export const OuiSplitButton = ({
+  color = 'primary',
+  fullWidth = false,
+  disabled,
+  options = [],
+  selectedIndex,
+  onSelection,
+  initiallyOpen = false,
+  hasDividers,
+  itemClassName,
+  onClick,
+  className,
+  popoverClassName,
+  children,
+  propsForDropdownButton,
+  propsForOptionItems,
+  propsForPrimaryButton,
+  ...rest
+}: OuiSplitButtonProps) => {
+  const [isOpen, setIsOpen] = useState(!!initiallyOpen);
+
+  const onSelectKeyDown = (
+    event: ExclusiveUnion<
+      React.KeyboardEvent<HTMLButtonElement>,
+      React.KeyboardEvent<HTMLAnchorElement>
+    >
+  ) => {
+    if (event.key === keys.ARROW_UP || event.key === keys.ARROW_DOWN) {
+      event.preventDefault();
+      event.stopPropagation();
+      setIsOpen(true);
+    }
+  };
+
+  const popoverClasses = classNames('ouiSplitButton', popoverClassName);
+
+  const buttonClasses = classNames(
+    {
+      'ouiSplitButton--isOpen__button': isOpen,
+    },
+    className
+  );
+
+  const itemClasses = classNames(
+    'ouiSplitButton__item',
+    {
+      'ouiSplitButton__item--hasDividers': hasDividers,
+    },
+    itemClassName
+  );
+
+  const button = (
+    <OuiSplitButtonControl
+      color={color}
+      onDropdownClick={() => setIsOpen(!isOpen)}
+      onClick={onClick}
+      onKeyDown={onSelectKeyDown}
+      className={buttonClasses}
+      fullWidth={fullWidth}
+      disabled={disabled}
+      propsForDropdownButton={propsForDropdownButton}
+      propsForPrimaryButton={propsForPrimaryButton}
+      {...rest}>
+      {children}
+    </OuiSplitButtonControl>
+  );
+
+  const textColor = (color: OuiSplitButtonColor) => {
+    if (color === 'primary') return 'success';
+    if (color === 'text') return 'default';
+
+    return color;
+  };
+
+  const items = options.map((option, index) => {
+    const isSelected = selectedIndex === index;
+
+    const content =
+      typeof option.display === 'string' ? (
+        <OuiText color={textColor(color)}>{option.display}</OuiText>
+      ) : (
+        option.display
+      );
+
+    return (
+      <OuiContextMenuItem
+        key={`optionItem_${index}`}
+        className={itemClasses}
+        color={color}
+        icon={isSelected ? 'check' : 'empty'}
+        href={option.href}
+        target={option.target}
+        onClick={option.onClick}
```

Comment:
The option click actions shouldn't occur here. They should be put on the primary button so I am able to select whichever button I want to use and perform the operation in 2 separate steps.

Reply:
Spoke offline. This is fine since the action thats performed on click is determined by the parent component, both switching options and immediately executing the option are possible like this.